lib/pdn/pdn.c

```diff
@@ -674,153 +680,81 @@ static int pdn_sa_family_from_ip_string(const char *src)
 {
 	char buf[INET6_ADDRSTRLEN];
 
-	if (inet_pton(AF_INET, src, buf)) {
+	if (zsock_inet_pton(AF_INET, src, buf)) {
 		return AF_INET;
-	} else if (inet_pton(AF_INET6, src, buf)) {
+	} else if (zsock_inet_pton(AF_INET6, src, buf)) {
 		return AF_INET6;
 	}
 	return -1;
 }
 
-int pdn_dynamic_info_get(uint32_t cid, struct pdn_dynamic_info *pdn_info)
+/** @brief Fill PDN dynamic info with DNS addresses adnd mtu. */
+static void pdn_dynamic_info_dns_addr_fill(struct pdn_dynamic_info *pdn_info, uint32_t mtu,
+					 const char *dns_addr_str_primary,
+					 const char *dns_addr_str_secondary)
 {
-	int ret = 0;
-	struct at_parser parser;
-	size_t param_str_len;
-
-	char *cgcontrdp_at_rsp_buf;
-
-	char *at_ptr;
-	char *tmp_ptr;
-	int lines = 0;
-	int iterator = 0;
-	char dns_addr_str[AT_CMD_PDN_CONTEXT_READ_IP_ADDR_STR_MAX_LEN];
-
-	char at_cmd_pdn_context_read_info_cmd_str[15];
-
 	int family;
-	struct in_addr *addr;
-	struct in6_addr *addr6;
 
-	if (!pdn_info) {
-		return -EINVAL;
-	}
-
-	cgcontrdp_at_rsp_buf = k_malloc(CONFIG_PDN_DYNAMIC_INFO_AT_BUF_SIZE);
-	if (!cgcontrdp_at_rsp_buf) {
-		LOG_ERR("k_malloc failed for CGDCONTRDP response buffer");
-		return -ENOMEM;
+	family = pdn_sa_family_from_ip_string(dns_addr_str_primary);
+	if (family == AF_INET) {
+		(void)zsock_inet_pton(AF_INET, dns_addr_str_primary,
+				      &(pdn_info->dns_addr4_primary));
+		(void)zsock_inet_pton(AF_INET, dns_addr_str_secondary,
+				      &(pdn_info->dns_addr4_secondary));
+		pdn_info->ipv4_mtu = mtu;
+	} else if (family == AF_INET6) {
+		(void)zsock_inet_pton(AF_INET6, dns_addr_str_primary,
+				      &(pdn_info->dns_addr6_primary));
+		(void)zsock_inet_pton(AF_INET6, dns_addr_str_secondary,
+				      &(pdn_info->dns_addr6_secondary));
+		pdn_info->ipv6_mtu = mtu;
 	}
+}
 
-	at_ptr = cgcontrdp_at_rsp_buf;
-	tmp_ptr = cgcontrdp_at_rsp_buf;
-
-	sprintf(at_cmd_pdn_context_read_info_cmd_str, AT_CMD_PDN_CONTEXT_READ_INFO, cid);
-	ret = nrf_modem_at_cmd(cgcontrdp_at_rsp_buf, CONFIG_PDN_DYNAMIC_INFO_AT_BUF_SIZE, "%s",
-			       at_cmd_pdn_context_read_info_cmd_str);
-	if (ret) {
-		LOG_ERR(
-			"nrf_modem_at_cmd returned err: %d for %s",
-			ret,
-			at_cmd_pdn_context_read_info_cmd_str);
-		goto clean_exit;
-	}
+int pdn_dynamic_info_get(uint8_t cid, struct pdn_dynamic_info *pdn_info)
+{
+	int ret;
+	char at_cmd_buf[sizeof("AT+CGCONTRDP=###")];
 
-	/* Check how many rows of info do we have */
-	while (strncmp(tmp_ptr, "OK", 2) &&
-	       (tmp_ptr = strstr(tmp_ptr, AT_CMD_PDN_CONTEXT_READ_RSP_DELIM)) != NULL) {
-		tmp_ptr += 2;
-		lines++;
-	}
+	char dns_addr_str_primary[INET6_ADDRSTRLEN];
+	char dns_addr_str_secondary[INET6_ADDRSTRLEN];
+	unsigned int mtu = 0;
```

Comment:
For consistency, use the same type name as the `ipv*_mtu` fields of `struct pdn_dynamic_info`.
```suggestion
	uint32_t mtu = 0;
```